include/http_ssl.h

```diff
@@ -49,6 +51,40 @@ AP_DECLARE_HOOK(int,ssl_conn_is_ssl,(conn_rec *c))
  */
 AP_DECLARE(int) ap_ssl_conn_is_ssl(conn_rec *c);
 
+/**
+ * This hook declares a connection to be outgoing and the configuration that applies to it.
+ * This hook can be called several times in the lifetime of an outgoing connection, e.g.
+ * when it is re-used in different request contexts. It will at least be called after the
+ * connection was created and before the pre-connection hooks is invoked.
+ * All outgoing-connection hooks are run until one returns something other than ok or decline.
+ * if enable_ssl != 0, a hook that sets up SSL for the connection needs to return DONE.
+ *
+ * @param c The connection on which requests/data are to be sent.
+ * @param dir_conf The directory configuration in which this connection is being used.
+ * @param enable_ssl If != 0, the SSL protocol should be enabled for this connection.
+ * @return OK or DECLINED, DONE when ssl was enabled
+ */
```

Comment:
First of all sorry for missing this on #190.
This is a run first hook. Hence it stops when the hook participant returns anything else but DECLINED.
1. What should be the difference between DONE and OK then? OK says I did not enable SSL, but I wanted to stop the hook?
2. The hook implementation in mod_ssl returns OK and not done when it enabled SSL. And `ap_ssl_bind_outgoing` checks for `OK` and not `DONE`
I think the fix is to adjust the documentation (`OK` signals enabled, `DECLINED` signals nothing done) here and leave the code as is.

Reply:
You are right. I will update the documentation.

Reply:
Resolved by update, I think.

Reply:
Thanks. It is resolved for the backport, but the old documentation is still in trunk